plugins/qcheck-stm/src/ir_of_gospel.ml

```diff
@@ -76,27 +76,35 @@ let ty_var_substitution config (vd : val_description) =
   let value_name, value_type = (vd.vd_name.id_str, vd.vd_type) in
   assert (is_a_function value_type);
   let open Ppxlib in
+  let open Reserr in
   let ret = function None -> Reserr.ok [] | Some x -> Reserr.ok x in
+  let check_sut_in_type_var ty =
+    match ty.ptyp_desc with
+    | Ptyp_constr (_, args) when List.exists (fun t -> Cfg.is_sut config t) args
```

Comment:
This doesn't check deep enough. For example it won't detect `sut list list`.

Reply:
True, I will adapt that!